File: 0257-binary-tree-paths/0257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> ans;
        if(root->left != NULL){
            vector<string> left = binaryTreePaths(root->left);
            for(int i = 0; i < left.size(); i++){
                string tmp = to_string(root->val);
                tmp += "->" + left[i];
                ans.push_back(tmp);
            }
        }
        if(root->right != NULL){
            vector<string> right = binaryTreePaths(root->right);
            for(int i = 0; i < right.size(); i++){
                string tmp = to_string(root->val);
                tmp += "->" + right[i];
                ans.push_back(tmp);
            }
        }
        if(ans.size() == 0){
            ans.push_back(to_string(root->val));
        }
        return ans;
    }
};

File: 0257-binary-tree-paths/0257-binary-tree-paths.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<string> BinaryTreePaths(TreeNode root) {
        //  https://leetcode.com/problems/binary-tree-paths/solutions/3013848/using-reccursive-approach
        List<string> paths = new List<string>();
        string s = "";
        FindPath(root, s);
        return paths;

        void FindPath(TreeNode root, string s1){
            //  현재 노드가 null 이 아닐때
            if(root != null){
                //  왼쪽, 오른쪽 자식 노드가 둘 다 없으면
                if(root.left == null && root.right == null){
                    //  현재 노드의 값을 문자열에 더하고
                    s1 = s1 + root.val;
                    //  리스트에 추가
                    paths.Add(s1);
                }else{
                    //  왼쪽, 오른쪽 자식 노드가 하나라도 있으면
                    //  현재 노드의 값과 화살표를 문자열에 더하고
                    //  왼쪽, 오른쪽 자식 노드로 이동
                    s1 = s1 + root.val + "->";
                    FindPath(root.left, s1);
                    FindPath(root.right, s1);
                }
            }
        }
    }

    
}